posts/src/main/java/com/collreach/posts/service/impl/PostServiceImpl.java

```diff
@@ -192,6 +192,13 @@ public MessagesResponse getPostsAndPollsPaginationFilteredByVisibility(Integer p
     @Override
     public LinkedHashSet<MessageResponse> mergeSets(LinkedHashSet<MessageResponse> posts, LinkedHashSet<MessageResponse> polls){
         LinkedHashSet<MessageResponse> mergedSets = new LinkedHashSet<>();
+        if(posts.size() == 0 && polls.size() == 0)
+            return mergedSets;
+        else if(posts.size() == 0)
+            return polls;
+        else if(polls.size() == 0)
+            return posts;
+
```

Comment:
This check was missing earlier due to which requests were giving Error 500 (Internal Server Error)